sound/soc/sof/topology.c

```diff
@@ -1758,17 +1802,23 @@ static int sof_widget_unload(struct snd_soc_component *scomp,
 	switch (swidget->id) {
 	case snd_soc_dapm_dai_in:
 	case snd_soc_dapm_dai_out:
-		dai = swidget->private;
+		dai = (struct snd_sof_dai *)swidget->private;
 
-		/* free dai config */
-		kfree(dai->dai_config);
-
-		/* remove and free dai object */
 		if (dai) {
+			/* free dai config */
+			kfree(dai->dai_config);
 			list_del(&dai->list);
-			kfree(dai);
 		}
 		break;
+	case snd_soc_dapm_scheduler:
+
+		/* disable pipeline schedule core */
+		pipeline = (struct sof_ipc_pipe_new *)swidget->private;
+		ret = snd_sof_dsp_core_power_down(sdev, 1 << pipeline->core);
+		if (ret < 0)
+			dev_err(sdev->dev, "error: powering down pipeline schedule core %d\n",
+				pipeline->core);
+		break;
```

Comment:
this patch should really split in 
a) addition of snd_sof_load_ipc
b) actions with snd_soc_dapm_scheduler on widget_unload
c) actions with snd_soc_dapm_scheduler on suspend, if any (add a comment if not needed)

Reply:
@plbossart will re-do the PR to fix them all